File: proj.android-studio/app/src/org/cocos2dx/cpp/ALGameHelpUtil.java
package org.cocos2dx.cpp;

import android.util.Log;

/**
 * Created by allen on 2017/12/26.
 */

public class ALGameHelpUtil {
    /**
     *  设置游戏用户的uid，必须在进入游戏前进行设置
     * @param uid
     */
    public static native  void setLoginUidOfGame(int uid);

    /**
     *  游戏分享成功后调用的方法
     *  此方法用于告诉游戏服务器分享成功，发放轻豆
     */
    public static native  void gameShareSuccessCallback();


    /**
     *  游戏主界面中的娘卡详情页面 点击分享按钮的相关操作
     *  用户需要实现此方法，应该在此方法中掉用相应的分享操作
     */
    public static void shareOfGameCardInfo()
    {
        // TODO: 2017/12/26  在此处写相关分享的方法
        Log.e("ALGameHelpUtil","游戏主界面的分享操作");
    }

    /**
     *   游戏内抽卡界面 中分享按钮的相关操作
     *  用户需要实现此方法，应该在此方法中掉用相应的分享操作
     */
    public static void shareOfGameDrawCard(int cardIndex)
    {
        // TODO: 2017/12/26 在此处写相关分享的操作
        Log.e("ALGameHelpUtil","游戏中的分享操作 卡片index = "+cardIndex);
    }

    /**
     *  退出游戏的相关一些操作  (此处用户点击退出游戏时候的一些操作)
     *  用户需要实现此方法，此方法中需要调用切换activity
     */
    public static void exitGame()
    {
        // TODO: 2017/12/26 实现退出游戏的相关操作
        Log.e("ALGameHelpUtil","退出游戏操作");

    }

}

File: Classes/ALGameHelpUtil.cpp
//
//  ALGameHelpUtil.cpp
//  LLKGame
//
//  Created by Allen on 2017/12/26.
//
//

#include "ALGameHelpUtil.h"
#include "cocos2d.h"

#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
#include "ALGameHelpUtilForAndroid.h"
#elif (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
#include "ALGameHelpUtilForIos.h"
#endif

static ALGameHelpUtil* s_instance = nullptr;
ALGameHelpUtil* ALGameHelpUtil::getInstance()
{
    if (!s_instance) {
        s_instance = new (std::nothrow) ALGameHelpUtil();
        CCASSERT(s_instance, "ALGameHelpUtil new Failed,not enoug memory");
    }
    return s_instance;
}


/**
 *  游戏主界面内的分享按钮的点击相关操作
 */
void ALGameHelpUtil::shareOfGameCardInfo()
{
#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
    ALGameHelpUtilForAndroid::getInstance()->shareOfGameCardInfo();
#elif (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
    ALGameHelpUtilForIos::getInstance()->shareOfGameCardInfo();
#endif
}

/**
 *  游戏中 抽卡分享按钮的点击相关操作
 */
void ALGameHelpUtil::shareOfGameDrawCard(int cardIndex)
{
#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
    ALGameHelpUtilForAndroid::getInstance()->shareOfGameDrawCard(cardIndex);
#elif (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
    ALGameHelpUtilForIos::getInstance()->shareOfGameDrawCard(cardIndex);
#endif
}

/**
 *  退出游戏的相关操作
 */
void ALGameHelpUtil::exitGame()
{
#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
    ALGameHelpUtilForAndroid::getInstance()->exitGame();
#elif (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
    ALGameHelpUtilForIos::getInstance()->exitGame();
#endif
}
